telemetry/client_test.go

```diff
@@ -296,13 +296,14 @@ func TestRetryCacheCleanup(t *testing.T) {
 
 	time.Sleep(110 * time.Millisecond)
```

Comment:
I think we should get rid of `time.Sleep()`. Tests must be deterministic.
Even if it works now, you might change smth in `Client` in the future that will make this test flaky again.

Reply:
Updated in latest commit. Added this issue for improving other telemetry tests: https://github.com/status-im/status-go/issues/5886